src/mmw/apps/user/middleware.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+
+from django.conf import settings
+from django.shortcuts import redirect
+
+from apps.user.models import ItsiUser
+
+URL_FLAG = settings.ITSI['url_flag']
+LOGIN_URL = settings.ITSI['login_url']
+
+
+class ItsiAuthenticationMiddleware(object):
+    """
+    Middleware for automatically logging in ITSI users
+    and setting relevant flags.
+    """
+
+    def process_request(self, request):
+        """
+        Check if ITSI URL FLAG is set, and if so attempt to log
+        the user in with their ITSI credentials and redirect them
+        to current page.
+        """
+
+        # If flag is not set then return None so Django can proceed
+        # with the request as-is
+        if request.GET.get(URL_FLAG, 'false') != 'true':
+            return None
+
+        # Check if user is already logged in, and if they are an
+        # ITSI user. If so, return None so Django can proceed
+        user = request.user
+        if (user.is_authenticated() and
+                ItsiUser.objects.filter(user_id=user.id)):
+            return None
+
+        # Flag is set and the user is not already logged in.
+        # Redirect them to ITSI LOGIN URL, with the current URL to return to
+        return redirect('{0}?next={1}'.format(LOGIN_URL, request.path))
```

Comment:
Is it okay to redirect from strings or should we use something more explicitly defined like `request.build_absolute_uri(reverse(...))` or something?